pkgs/by-name/vi/video2x/package.nix

```
@@ -0,0 +1,97 @@
+{
+  lib,
+  stdenv,
+  cmake,
+  fetchFromGitHub,
+  pkg-config,
+  versionCheckHook,
+  gitUpdater,
+  boost,
+  ffmpeg,
+  glslang,
+  llvmPackages,
+  ncnn,
+  shaderc,
+  spdlog,
+  vulkan-headers,
+  vulkan-loader,
+  vulkan-tools,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "video2x";
+  version = "6.4.0";
+
+  src = fetchFromGitHub {
+    owner = "k4yt3x";
+    repo = "video2x";
+    tag = finalAttrs.version;
+    hash = "sha256-DSsfGAkPOtqqj0FA1N33O+OYmv+CMUsrvmh5SrnF7eA=";
+    fetchSubmodules = false;
+    leaveDotGit = true;
+    postFetch = ''
+      pushd $out
+      git reset --hard HEAD
+
+      # Fetch the dependencies that nixpkgs cannot provide (non-recursive)
+      git submodule update --init third_party/librealesrgan_ncnn_vulkan
+      git submodule update --init third_party/librealcugan_ncnn_vulkan
+      git submodule update --init third_party/librife_ncnn_vulkan
+
+      # Cleanup
+      rm -rf .git
+
+      popd
+    '';
+  };
+
+  postPatch = ''
+    substituteInPlace src/fsutils.cpp \
+      --replace-fail '/usr/share/video2x' '${placeholder "out"}/share/video2x'
+  '';
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    glslang
+  ];
+
+  buildInputs =
+    [
+      boost
+      ffmpeg.dev
```

```suggestion
      ffmpeg
```

`.dev` is automatically included by stdenv.